src/internet_identity_interface/src/http_gateway.rs

```diff
@@ -0,0 +1,36 @@
+use candid::{CandidType, Deserialize, Func};
+use serde_bytes::{ByteBuf, Bytes};
+use std::borrow::Cow;
+
+pub type HeaderField = (String, String);
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct Token {}
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub enum StreamingStrategy {
+    Callback { callback: Func, token: Token },
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct StreamingCallbackHttpResponse {
+    pub body: ByteBuf,
+    pub token: Option<Token>,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct HttpRequest {
+    pub method: String,
+    pub url: String,
+    pub headers: Vec<(String, String)>,
+    pub body: ByteBuf,
+}
+
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct HttpResponse {
+    pub status_code: u16,
+    pub headers: Vec<HeaderField>,
+    pub body: Cow<'static, Bytes>,
+    pub upgrade: Option<bool>,
```

Comment:
What is the meaning of the flag, i.e. what does it mean when it is true vs. when it is false? Will  in the future some value be assumed as a default when the value is `None`?  Is it worth documenting the semantics in a comment?  (maybe the meaning is clear for a knowledgable reader, so sorry if the question is stupid)

Reply:
Given the semantics of these types are quite complex (also with regards to streaming) I'd rather just delegate to the spec that explains all those fields.
So I suggest to also add a link to the spec in this file.